### PR TITLE
[M2-2583] Feature: events sync

### PR DESCRIPTION
**Pull request by moiskillnadne**

### CODE REVIEW

File: src/widgets/ActivityGroups/model/hooks/useCompletedEntitiesBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            type: ActivityPipelineType.Regular,
+            startAt: null,
+            endAt: new Date(endAtDate),
+          },
+        })
+      } else {
+        if (inProgressEvent.endAt) {
+          const isServerEndAtBigger = endAtDate > new Date(inProgressEvent.endAt).getTime()
+
+          if (isServerEndAtBigger) {
```
Comment (BamMironov) [inline src/widgets/ActivityGroups/model/hooks/useCompletedEntitiesBuilder.ts:46]: I am almost sure that we can get rid of nested ifs

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/widgets/ActivityGroups/model/hooks/useCompletedEntitiesBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          }
+        }
+      }
+    })
+  }, [
+    groupsInProgress,
+    props.appletId,
+    props.completedEntities,
+    props.completedEntities?.activities,
+    props.completedEntities?.activityFlows,
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/widgets/ActivityGroups/model/hooks/useCompletedEntitiesBuilder.ts:65]: Done

File: src/widgets/ActivityGroups/model/hooks/useCompletedEntitiesBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+              eventId: entity.scheduledEventId,
+              progressPayload: {
+                ...inProgressEvent,
+                endAt: new Date(endAtDate),
+              },
+            })
+          }
+        }
+      }
+    })
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/widgets/ActivityGroups/model/hooks/useCompletedEntitiesBuilder.ts:59]: Done

File: src/widgets/ActivityGroups/model/hooks/useCompletedEntitiesBuilder.ts (hunk 1 above)
Comment (moiskillnadne) [reply src/widgets/ActivityGroups/model/hooks/useCompletedEntitiesBuilder.ts:46]: Done

**Review by BamMironov — APPROVED** [review]

**Pull request merged**